src/elm/Pages/Build/Graph/DOT.elm

```diff
@@ -0,0 +1,510 @@
+                ++ "</font></td>"
+                ++ "</tr>"
+
+        link step =
+            Routes.routeToUrl <|
+                Routes.Build repo.org
+                    repo.name
+                    (String.fromInt build.number)
+                    (Just <|
+                        Focus.resourceFocusFragment
+                            "step"
+                            (String.fromInt step.number)
+                            []
+                    )
+
+        -- table row and cell styling
+        rowAttributes _ =
+            [-- row attributes go here
+            ]
+
+        cellAttributes step =
+            [ ( "border", DefaultEscape "0" )
+            , ( "cellborder", DefaultEscape "0" )
+            , ( "cellspacing", DefaultEscape "0" )
+            , ( "margin", DefaultEscape "0" )
+            , ( "align", DefaultEscape "left" )
+            , ( "href", DefaultEscape <| link step )
+            , ( "id", DefaultEscape "node-cell" )
+            , ( "title", DefaultEscape ("#status-" ++ statusToString step.status) )
+            , ( "tooltip", DefaultEscape (String.join "," [ String.fromInt step.id, step.name, statusToString step.status ]) )
+            ]
+
+        row step =
+            "<tr "
+                ++ escapeAttributes
+                    (rowAttributes step)
+                ++ ">"
+                ++ "<td "
+                ++ escapeAttributes
+                    (cellAttributes step)
+                ++ " "
+                ++ ">"
+                -- required icon spacing
+                ++ "     "
+                ++ "<font>"
+                ++ step.name
+                ++ "</font>"
+                ++ "<br align='left'/>"
+                ++ "</td>"
+                ++ "</tr>"
+
+        rows =
+            if showSteps then
+                List.map row steps
+
+            else
+                []
+    in
+    table <| header :: rows
+
+
+{-| nodeLabel : takes model and a node, and returns the DOT string representation
+-}
+nodeToString : BuildModel.PartialModel a -> Repository -> Build -> Node BuildGraphNode -> String
+nodeToString model repo build node =
+    "  "
+        ++ String.fromInt node.id
+        ++ makeAttributes (nodeAttributes model repo build node.label)
+
+
+{-| edgeToString : takes model and a node, and returns the DOT string representation
+-}
+edgeToString : Edge BuildGraphEdge -> String
+edgeToString edge =
+    "  "
+        ++ String.fromInt edge.from
+        ++ " -> "
+        ++ String.fromInt edge.to
+        ++ makeAttributes (edgeAttributes edge.label)
+
+
+
+-- STYLES
+
+
+{-| baseGraphStyles : returns the base styles applied to the root graph.
+-}
+baseGraphStyles : Rankdir -> Styles
+baseGraphStyles rankdir =
+    { rankdir = rankdir
+    , graph =
+        escapeAttributes
+            [ ( "bgcolor", DefaultEscape "transparent" )
+            , ( "splines", DefaultEscape "ortho" )
+
+            -- , ( "compound", DefaultEscape "true" )
```

Comment:
do we need to keep this?

Reply:
oh, just saw this, no, i was just playing around with compound graphs. let me remove that before it ends up being there forever